File: SSM/SSM_PET/src/java/SSM_PET/petExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class petExogenous
{
    private double tmax;
    private double tmin;
    private double srad;
    
    public petExogenous() { }
    
    public petExogenous(petExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.tmax = toCopy.gettmax();
            this.tmin = toCopy.gettmin();
            this.srad = toCopy.getsrad();
        }
    }
    public double gettmax()
    { return tmax; }

    public void settmax(double _tmax)
    { this.tmax= _tmax; } 
    
    public double gettmin()
    { return tmin; }

    public void settmin(double _tmin)
    { this.tmin= _tmin; } 
    
    public double getsrad()
    { return srad; }

    public void setsrad(double _srad)
    { this.srad= _srad; } 
    
}

File: SSM/SSM_PET/src/cpp/SSM_PET/petExogenous.cpp
#include "petExogenous.h"
using namespace SSM_PET;


petExogenous::petExogenous() {}

double petExogenous::gettmax() { return this->tmax; }
double petExogenous::gettmin() { return this->tmin; }
double petExogenous::getsrad() { return this->srad; }

void petExogenous::settmax(double _tmax) { this->tmax = _tmax; }
void petExogenous::settmin(double _tmin) { this->tmin = _tmin; }
void petExogenous::setsrad(double _srad) { this->srad = _srad; }